staging/src/k8s.io/apiserver/pkg/endpoints/handlers/proxy.go

```diff
@@ -283,3 +274,14 @@ func singleJoiningSlash(a, b string) string {
 	}
 	return a + b
 }
+
+func RecordMetrics(w http.ResponseWriter, req *http.Request, requestInfo *request.RequestInfo, httpCode int, reqStart time.Time) {
```

Comment:
Compare https://github.com/kubernetes/kubernetes/blob/fabce1b893f96bdf466c1fdb1fcf825210c008ae/staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go#L151. Can you move this there and align the signature?

Reply:
Any blockers here?

Reply:
> Any blockers here?

We don't have a `restful.RouteFunction` here in this handler. Any reason why `RecordMetrics` is incorrect?

Reply:
@sttts

Reply:
@sttts updated to use `InstrumentHandlerFunc`. ptal, thanks.